src/goal_setter.cpp

```diff
@@ -0,0 +1,138 @@
+#include "ros/ros.h"
+#include "hide_and_seek/tp_msg.h"
+#include "hide_and_seek/tp_map_srv.h"
+#include "nav_msgs/OccupancyGrid.h"//PICK AN ACTUAL MAP MESSAGE YO MAYBE OCCUPANCY GRID??
+#include "nav_msgs/GetMap.h"
+#include "move_base_msgs/MoveBaseAction.h"
+#include <actionlib/client/simple_action_client.h>
+#include <math.h>
+
+#define PROXIMITY_THRESHOLD 4
+
+class GoalSetter//Map Reciever and Tourist Point List Generator
+{
+
+public:
+  GoalSetter()
+  {
+//tpl_service = n.advertiseService("generate_new_tp_list", generate_new_tpl);
+    
+    sub = n.subscribe("amcl_pose", 10, &GoalSetter::current_pos_retriever, this);
+    first_spin = true;
+    goal_index = -1;
+    ros::spinOnce();  //spun for her pleasure..... and to get the initial position cause set_goal needs it
+    tpmap_reciever = n.serviceClient<hide_and_seek::tp_map>("generate_new_tp_list");  //this cant be right
+    //set_goalster = n.advertiseService("")
+    action_client("move_base", false);  //I'm not positive whether this should be true
+    while(!action_client.waitForServer( ros::Duration(5.0) ))
+    {
+      ROS_INFO("Waiting for the move_base action server to come up");
+    }  //in constructor so it doesnt wait 5 sec after every tourist point
+    //ottpm.request.postcard = false;
+    person_alert = n.subscribe("person_present", 10);
+    this.set_goal();
+  }
+      //prototypes bb
+  void GoalSetter::set_goal(hide_and_seek::tp_map_srv::Request &req, hide_and_seek::tp_map_srv::Response &res);
+  void current_pos_retriever(const geometry_msgs::PoseWithCovarianceStamped::ConstPtr& current_pos);
+  void GoalSetter::person_alert_callback(const std_msgs::Bool::ConstPtr& person_present_msg)
+  
+private:
+  ros::NodeHandle n;
+  ros::ServiceClient tpmap_reciever;
+  ros::Publisher goal_pub;
+  geometry_msgs::PoseWithCovarianceStampedMessage current_position;
+  //ros::ServiceClient current_position;
+  ros::Subscriber sub;
+  ros::Subscriber person_alert;
+  actionlib::SimpleActionClient<move_base_msgs::MoveBaseAction> action_client;
+  move_base_msgs::MoveBaseGoal current_goal;
+  hide_and_seek::tp_map_srv ottpm;  //The one true tourist point map
+  bool first_spin;
+  int goal_index;
+  //ros::ServiceServer set_goalster;
```

Comment:
there's a lot of commented out code in this file. Please remove it before merge.

Reply:
Yeah, its legit like a minefield of fallen code, never took the time to do that, but I will now